presto-hive/src/test/java/com/facebook/presto/hive/TestHiveCommitHandleOutput.java

```diff
@@ -0,0 +1,580 @@
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+
+import static com.facebook.airlift.concurrent.Threads.daemonThreadsNamed;
+import static com.facebook.presto.common.type.BigintType.BIGINT;
+import static com.facebook.presto.hive.AbstractTestHiveClient.TEST_SERVER_VERSION;
+import static com.facebook.presto.hive.HiveStorageFormat.ORC;
+import static com.facebook.presto.hive.HiveTableProperties.BUCKETED_BY_PROPERTY;
+import static com.facebook.presto.hive.HiveTableProperties.BUCKET_COUNT_PROPERTY;
+import static com.facebook.presto.hive.HiveTableProperties.EXTERNAL_LOCATION_PROPERTY;
+import static com.facebook.presto.hive.HiveTableProperties.PARTITIONED_BY_PROPERTY;
+import static com.facebook.presto.hive.HiveTableProperties.SORTED_BY_PROPERTY;
+import static com.facebook.presto.hive.HiveTableProperties.STORAGE_FORMAT_PROPERTY;
+import static com.facebook.presto.hive.HiveTestUtils.FILTER_STATS_CALCULATOR_SERVICE;
+import static com.facebook.presto.hive.HiveTestUtils.FUNCTION_AND_TYPE_MANAGER;
+import static com.facebook.presto.hive.HiveTestUtils.FUNCTION_RESOLUTION;
+import static com.facebook.presto.hive.HiveTestUtils.ROW_EXPRESSION_SERVICE;
+import static com.facebook.presto.hive.metastore.MetastoreUtil.PRESTO_QUERY_ID_NAME;
+import static com.facebook.presto.hive.metastore.MetastoreUtil.toPartitionValues;
+import static com.facebook.presto.hive.metastore.StorageFormat.fromHiveStorageFormat;
+import static java.nio.file.Files.createTempDirectory;
+import static java.util.concurrent.Executors.newFixedThreadPool;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertTrue;
+
+public class TestHiveCommitHandleOutput
+{
+    private static final String TEST_SCHEMA = "test_schema";
+    private static final String TEST_TABLE = "test_table";
+
+    private static Map<String, Object> testTableProperties;
+    private static ConnectorTableMetadata testTableMetadata;
+
+    private ListeningExecutorService listeningExecutor;
+
+    @BeforeClass
+    public void setUp()
+            throws IOException
+    {
+        URI tempUri = createTempDirectory("test").toUri();
+
+        ImmutableMap.Builder<String, Object> builder = ImmutableMap.builder();
+        builder.put(BUCKET_COUNT_PROPERTY, 0)
+                .put(BUCKETED_BY_PROPERTY, ImmutableList.of())
+                .put(SORTED_BY_PROPERTY, ImmutableList.of())
+                .put(STORAGE_FORMAT_PROPERTY, ORC)
+                .put(EXTERNAL_LOCATION_PROPERTY, tempUri.toASCIIString())
+                .put(PARTITIONED_BY_PROPERTY, ImmutableList.of("a"));
+        testTableProperties = builder.build();
+
+        testTableMetadata = new ConnectorTableMetadata(
```

Comment:
is there any reason not to make these static final variables instead of constructing them in a setup method?

Reply:
So far no, we can make them as static final.

Reply:
Done